File: Hollow Rectangle/printHollowRectangle.py
rows = int(input("Enter the number of rows : "));
columns = int(input("Enter the number of columns : "));

for i in range(0, rows):
  for j in range(0, columns):
    if (i==0 or j==0 or i==rows-1 or j==columns-1):
      print("*", end = "")
    else:
      print(" ", end = "")
  print()

File: Hollow Rectangle/printHollowRectangle.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("Enter the number of rows : ");
    int rows = sc.nextInt();
    System.out.println("Enter the number of columns : ");
    int columns = sc.nextInt();

    for (int i=0; i<rows; i++) {
      for (int j=0; j<columns; j++) {
        if (i==0 || j==0 || i==rows-1 || j==columns-1) {
          System.out.print("*");
        } else {
          System.out.print(" ");
        }
      }
      System.out.print("\n");
    }

    sc.close();
  }
}

